### PR TITLE
qbd-direct post onboarding configuration export settings

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
+import { FormControl, FormGroup } from "@angular/forms";
+import { ExportModuleRule, ExportSettingModel, ExportSettingValidatorRule } from "../../common/export-settings.model";
+import { CCCExpenseState, ExpenseState, QBDCorporateCreditCardExpensesObject, QBDEntity, QBDExpenseGroupedBy, QBDExportDateType, QBDReimbursableExpensesObject } from "../../enum/enum.model";
+import { QBDExportSettingFormOption } from "../../qbd/qbd-configuration/qbd-export-setting.model";
+
+export type QbdDirectExportSettingsPost = {
+    is_simplify_report_closure_enabled: boolean,
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:7]: not required

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    reimbursable_expenses_export_type: QBDReimbursableExpensesObject | null,
+    bank_account_name: string | null,
+    mileage_account_name : string | null,
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:10]: not required

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
+import { FormControl, FormGroup } from "@angular/forms";
+import { ExportModuleRule, ExportSettingModel, ExportSettingValidatorRule } from "../../common/export-settings.model";
+import { CCCExpenseState, ExpenseState, QBDCorporateCreditCardExpensesObject, QBDEntity, QBDExpenseGroupedBy, QBDExportDateType, QBDReimbursableExpensesObject } from "../../enum/enum.model";
+import { QBDExportSettingFormOption } from "../../qbd/qbd-configuration/qbd-export-setting.model";
+
+export type QbdDirectExportSettingsPost = {
+    is_simplify_report_closure_enabled: boolean,
+    reimbursable_expenses_export_type: QBDReimbursableExpensesObject | null,
+    bank_account_name: string | null,
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:9]: check API models and create type similar to it, this is not required

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    static cccExpenseStateOptions(is_simplify_report_closure_enabled: boolean): QBDExportSettingFormOption[] {
+        return [
+            {
+            label: 'Approved',
+            value: CCCExpenseState.APPROVED
+            },
+            {
+            label: is_simplify_report_closure_enabled ? 'Closed' : 'Paid',
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:103]: remove is_simplify_report_closure_enabled references, assume it's always true

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+            value: CCCExpenseState.PAID
+            }
+        ];
+    }
+
+    static expenseStateOptions(is_simplify_report_closure_enabled: boolean): QBDExportSettingFormOption[] {
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:109]: same as above

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+            value: QBDExportDateType.POSTED_AT
+          }]);
+        }
+        return [this.reimbursableExpenseGroupingDateOptions()[1]];
+    }
+
+    static getValidators(): [ExportSettingValidatorRule, ExportModuleRule[]] {
+
+        const exportSettingValidatorRule: ExportSettingValidatorRule = {
+          reimbursableExpense: ['reimbursableExportType', 'reimbursableExportGroup', 'reimbursableExportDate', 'reimbursableExpenseState', 'bankAccount', 'mileageAccountName'],
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:146]: remove 'bankAccount', 'mileageAccountName'

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+          creditCardExpense: ['creditCardExportType', 'creditCardExportGroup', 'creditCardExportDate', 'creditCardExpenseState', 'creditCardEntityName', 'creditCardAccount']
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:147]: remove 'creditCardEntityName', 'creditCardAccount'

File: src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+            creditCardExportType: new FormControl(exportSettings?.credit_card_expense_export_type ? exportSettings?.credit_card_expense_export_type : null),
+            creditCardExportGroup: new FormControl(exportSettings?.credit_card_expense_grouped_by ? exportSettings?.credit_card_expense_grouped_by : this.expenseGroupingFieldOptions()[1].value),
+            creditCardExportDate: new FormControl(exportSettings?.credit_card_expense_date ? exportSettings?.credit_card_expense_date : this.expenseGroupingFieldOptions()[0].value),
+            bankAccount: new FormControl(exportSettings?.bank_account_name ? exportSettings?.bank_account_name : null),
+            mileageAccountName: new FormControl(exportSettings?.mileage_account_name ? exportSettings?.mileage_account_name : null),
+            creditCardEntityName: new FormControl(exportSettings?.credit_card_entity_name_preference ? exportSettings?.credit_card_entity_name_preference : null),
+            creditCardAccount: new FormControl(exportSettings?.credit_card_account_name ? exportSettings?.credit_card_account_name : null),
```
Comment (ashwin1111) [inline src/app/core/models/qbd-direct/qbd-direct-configuration/qbd-direct-export-settings.model.ts:168]: not required, 165-168

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 9]
```diff
@@ -7,6 +21,133 @@ import { Component } from '@angular/core';
...
-export class QbdDirectExportSettingsComponent {
+export class QbdDirectExportSettingsComponent implements OnInit{
+
+  isLoading: boolean;
+
+  isOnboarding: any;
+
+  exportSettings: QbdDirectExportSettingGet | null;
+
+  is_simplify_report_closure_enabled: any;
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:32]: not required

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 10]
```diff
@@ -7,6 +21,133 @@ import { Component } from '@angular/core';
...
+    @Inject(FormBuilder) private formBuilder: FormBuilder,
+    // Private workspaceService: QbdWorkspaceService,
+    private toastService: IntegrationsToastService,
+    private trackingService: TrackingService,
+    public helperService: HelperService,
+    private mappingService: MappingService
+  ) { }
+
+  cccExportTypeWatcher(): void {
+    this.exportSettingsForm.controls.creditCardExportType.valueChanges.subscribe((creditCardExportTypeValue) => {
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:64]: will revisit all conditions once backend stuffs are cleared

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 11]
```diff
@@ -7,6 +21,133 @@ import { Component } from '@angular/core';
...
+    } else {
+      this.cccExpenseGroupingDateOptions = QbdDirectExportSettingModel.setCreditCardExpenseGroupingDateOptions(QBDCorporateCreditCardExpensesObject.CREDIT_CARD_PURCHASE, QBDExpenseGroupedBy.EXPENSE);
+    }
+  }
+
+  private getSettingsAndSetupForm(): void {
+    this.isLoading = true;
+    this.isOnboarding = this.router.url.includes('onboarding');
+
+    const destinationAttributes = ['BANK_ACCOUNT', 'CREDIT_CARD_ACCOUNT'];
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:114]: skip this for now

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts [hunk 12]
```diff
@@ -7,6 +21,133 @@ import { Component } from '@angular/core';
...
+    destinationAttributes.forEach((destinationAttribute) => {
+      groupedAttributes.push(this.mappingService.getPaginatedDestinationAttributes(destinationAttribute).pipe(filter(response => !!response)));
+    });
+
+    forkJoin([
+      this.exportSettingService.getQbdExportSettings()
+      // ...groupedAttributes
+    ]).subscribe(([exportSettingResponse]) => {
+      this.exportSettings = exportSettingResponse;
+      this.is_simplify_report_closure_enabled = this.exportSettings?.is_simplify_report_closure_enabled;
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.ts:127]: not required

**Comment by DhaaraniCIT** [comment]
PR comments are fixed in https://github.com/fylein/fyle-integrations-app/pull/947

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**